### ISSUE TITLE
Question: Why shouldn't Store be typed?

### ISSUE THREAD

**Issue by patoncrispy** [timdeschryver/eslint-plugin-ngrx#34]
I just added these rules to my codebase and it flagged a lot of errors which is great, but I don't understand the rule `no-typed-store`. Why shouldn't a store be typed? I've tried Googling it but struggling to understand this. 

Edit: If I remove the generics I get the following error:
```
Generic type 'Store<T>' requires 1 type argument(s)
```

**Comment by patoncrispy** [comment]
Thanks so much for that excellent explanation, @timdeschryver. Makes a lot of sense now and will help me tidy up our codebase a bit once we get to upgrade.

**Comment by urielzen** [comment]
I am sorry to ask a following question on a closed issue.
I am using ngrx 12.4.0 and getting this warning "Store should not be typed, use `Store` (without generic) instead"

my injected store was typed, and when trying to remove the type to fix this warning, I have compilation errors in my selectors.

My selector
```ts
import { createSelector } from '@ngrx/store';
import { AppState } from '../app.state';

export const colorThemeState = (state: AppState) => state.colorTheme;

export const selectColorThemeName = createSelector(
  colorThemeState,
  state => state.name
);
```

In my component
```ts
  constructor(
    private store: Store
  ) { }

  colorThemeName$ = this.store.select(selectColorThemeName); // Error here
```

The last line above on `this.store.select()` gives the following error:

```
No overload matches this call.
  Overload 1 of 9, '(mapFn: (state: object) => ColorTheme): Observable<ColorTheme>', gave the following error.
    Argument of type 'MemoizedSelector<AppState, ColorTheme, DefaultProjectorFn<ColorTheme>>' is not assignable to parameter of type '(state: object) => ColorTheme'.
      Types of parameters 'state' and 'state' are incompatible.
        Type '{}' is missing the following properties from type 'AppState': fetchingData, colorTheme
  Overload 2 of 9, '(key: never): Observable<never>', gave the following error.
    Argument of type 'MemoizedSelector<AppState, ColorTheme, DefaultProjectorFn<ColorTheme>>' is not assignable to parameter of type 'never'.ts(2769)

```

I imagine I am missing something obvious at this point, but I am unable to find what it is at this point.

Thanks in advance!

**Comment by timdeschryver** [comment]
@urielzen see #79
